sdk/tables/azure-data-tables/azure/data/tables/_error.py

```diff
@@ -61,9 +61,9 @@ def _wrap_exception(ex, desired_type):
 
 
 def _validate_table_name(table_name):
-    if match("^[a-zA-Z]{1}[a-zA-Z0-9]{2,62}$", table_name) is None:
+    if table_name is None:
```

Comment:
This is doing the same check for table_name as in [_table_client.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/azure/data/tables/_table_client.py#L78) and [_table_client_async.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py#L76), you can remove this function.

Reply:
Thanks @YalinLi0312 
I've checked again Storage Account - Table Service, the creation of Table requires its name matched the regex defined, otherwise users would receive the following errors.
Would it possible to move _validate_table_name to create_table and differentiate the process is for checking constrain of Blob Storage Table service or Cosmos DB - Table API? Since these two services have different limitations of their table name.
So that we can safely remove the checking when TableClient acquiring specified table name since it should already pass the checking when creating.

case 1: TableName = 1
> **The specified resource name length is not within the permissible limits.**
> RequestId:c906af43-5002-000d-2929-1c2332000000
> Time:2022-02-07T13:50:13.9464275Z
> ErrorCode:OutOfRangeInput

case 2: Table name = 123
> **The specifed resource name contains invalid characters.**
> RequestId:2ee4c645-e002-0032-3f29-1c8b3c000000
> Time:2022-02-07T13:50:52.6155078Z
> ErrorCode:InvalidResourceName

case 3: Table name = myTable-01
> **The specifed resource name contains invalid characters.**
> RequestId:f30cb0d0-8002-0024-262a-1cc112000000
> Time:2022-02-07T13:55:11.8494619Z
> ErrorCode:InvalidResourceName

Reply:
Thanks for the discussion.
Since the SDK faces both services (Cosmos DB -Table API, and Blob Storage - Table Service), thus the committed code changes aren't applicable.
We need to consider the naming rules for both, adding some details below for our tracking, 

[Cosmos DB - Resource name rules](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftdocumentdb)
No specified for container/table level rule in doc.
But [in other doc](https://docs.microsoft.com/en-us/azure/cosmos-db/sql/create-cosmosdb-resources-portal), explain the limitation of container/table name should be:
names must contain from 1 through 255 characters, and they cannot contain /, \\, #, ?, or a trailing space

[Storage Account -Resource name rules](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftstorage)
table / 3-63 / Alphanumerics + Start with letter

Reply:
Cancel [PR#22938](https://github.com/Azure/azure-sdk-for-python/pull/22938), create a new [Issue#23106](https://github.com/Azure/azure-sdk-for-python/issues/23106) for tracking this issue.